File: core/java/src/main/java/hara/lang/protocol/IReset.java
package hara.lang.protocol;

import hara.lang.declaration.HaraMethod;
import hara.lang.declaration.HaraProtocolBinding;

@HaraProtocolBinding(namespace = "std.protocol.ireset", name = "IReset")
public interface IReset<V> {
  @HaraMethod(value = "reset", arity = 2)
  V reset(V v);
}

File: core/rust/src/lang/protocol/ireset.rs
use hara_protocol_macros::hara_protocol;

#[hara_protocol(namespace = "std.protocol.ireset", name = "IReset")]
pub trait IReset<V> {
    type Error;

    #[hara_method(value = "reset", arity = 2)]
    fn reset(&self, value: V) -> Result<V, Self::Error>;
}
